### DISCUSSION TITLE
How to enable AutoRefresh

### DISCUSSION THREAD

**Discussion by ffwrude** [zanllp/infinite-image-browsing#617] [Q&A]
Hello,

Apparently this app supports autoRefresh since May 2023 but i don't know how to enable it. There's nothing in the settingsd tab.

**Comment by zanllp** [comment]
The feature is enabled by default and has no option to disable it.

**Comment by ffwrude** [comment]
Guess it doesn't work on my end then. Oh well... Not going to fiddle with it. Don't want to break it.

Télécharger Outlook pour Android<https://aka.ms/ghei36>
________________________________

**Comment by zanllp** [comment]
When this feature is triggered, the following message will be displayed, but due to limitations of the browser itself, sometimes refreshing may not work smoothly. You can manually refresh by clicking the refresh button in the top right corner.

![image](https://github.com/zanllp/sd-webui-infinite-image-browsing/assets/25872019/22e6f8c1-798e-487f-820c-f983986f88ec)

**Reply by lr45lr45** [reply]
I try in a few different browsers, and the auto-refresh never works for me. 
Images are being generated to an output folder (txt2img) but the only way to see them in the infinite image browser is to manually click refresh button.

Would be great if they could auto-refresh.

I see from previous messages there are no settings for auto-refresh.

Is there a browser setting that is needed to allow auto-refresh?
Does anyone know how it can be made to work?

**Reply by zanllp** [reply]
@lr45lr45 Is the auto-refresh not working properly in walk mode? If you are in walk mode, that's normal, as implementing auto-refresh in walk mode is quite complex, and I haven't implemented that feature yet. However, if you need it, I will try to implement it.

**Reply by lr45lr45** [reply]
Thanks for your help. The auto-refresh is not working in walk-mode.

It would be very useful to have the auto-refresh so we can see the differences between images as they are generated.

Just a suggestion, could the existing manual refresh be triggered in the background every x seconds by your code as an alternative workaround. Example every 3 seconds.  Thanks

**Reply by zanllp** [reply]
https://github.com/zanllp/sd-webui-infinite-image-browsing/wiki/Change-log#implemented-automatic-refresh-in-walk-mode

I have implemented this feature, although it appears to have performance issues under certain circumstances.

 Walk mode does not directly retrieve files from the database; instead, it maintains a tree of file nodes on the frontend and uses depth-first search (DFS) to generate the file list. This approach offers good versatility, as it can be applied to any folder using the walk mode and is also friendly to existing data. However, the downside is that the code becomes very complex. If a single folder contains too few files, the number of API requests could become very high, making automatic refresh in walk mode a challenging problem. This is especially true when the server is cloud-based or when scrolling to the very bottom of the page (a single walk mode page can easily accommodate tens of thousands of images), where performance might be significantly worse.

**Reply by lr45lr45** [reply]
Thanks. 
I updated the extension and restarted everything.
It does not seem to work for me.

Here is how I checked it:

i) started generating a text to image and set it to generate forever, ie, Right-click on the 'Generate' button and choose 'Generate Forever' 

ii) In IIB tab looked in walk mode and selected text2image and can see the previously generated images

iii) As new images are generated by the text2image 'Generate Forever'  they are not appearing in the IIB tab

My hope was that while images are generated they would automatically appear on the IIB tab so you can see them like a live feed.

**Reply by zanllp** [reply]
Which version of sd-webui are you using?

**Reply by lr45lr45** [reply]
My sd-webui is 1.9.4

When inside the txt2img, if I then click on the IIB tab then it does the refresh.
But that is not what I was hoping for.

I was hoping while in the IIB that it would update in real time as new images are generated by the txt2img "Generate Forever". 

Is the above is possible please?

**Comment by zanllp** [comment]
@lr45lr45 I tried it with version 1.9.4 of sd-webui, and even using "Generate Forever," IIB seems to work fine. Is your IIB not updated?


https://github.com/user-attachments/assets/3652679d-4d6f-402e-8708-6add5237ffca

**Reply by zanllp** [reply]
This is a notification for "Detected file change, auto-updating." Due to the significant computational load, unlike the normal mode, it also includes a loading indicator.

![image](https://github.com/user-attachments/assets/6a2812cb-539f-4eec-be02-2581cd188f63)

**Reply by lr45lr45** [reply]
I have the latest version of IIB.
Many thanks for making the video.

I think you are misunderstanding me. I'm sorry to say.

With the IIB open and visible, I expect new images to show in IIB as they are generated.

For example.
- Set generate forever in txt2img
- Go to the IIB tab
- New image 1 is created, and it then shows in IIB
- New image 2 is created, and it then shows in IIB
etc etc

Therefore, you can watch each image as it's generated within IIB.

Sorry if that was not made clear before.

**Reply by zanllp** [reply]
Ok, I understand now, that's what it means. I'll give it a try.

**Comment by zanllp** [comment]
@lr45lr45 <img width="1069" alt="image" src="https://github.com/user-attachments/assets/4e10fd90-b291-4282-a23a-309cc96855a4">

**Reply by lr45lr45** [reply]
That looks great, really appreciate that, thanks.
